Build failure: snes9x-gtk

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

- use unstable
- add snes9x-gtk to config
- rebuild

### Can Hydra reproduce this build failure?

Hydra doesn’t try to build the package.

### Link to Hydra build job

_No response_

### Relevant log output

```console
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/mkyha1pb28h8krnq8mjk9z1965wrasa8-source
source root is source
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Updating Autotools / GNU config script to a newer upstream version: ./unix/config.sub
Updating Autotools / GNU config script to a newer upstream version: ./unix/config.guess
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
cmake flags: -GNinja -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTING=OFF -DCMAKE_INSTALL_LOCALEDIR=/nix/store/a62mjw98vmi7xznabalpi5yjb43yvchn-snes9x-gtk-1.63/share/locale -DCMAKE_INSTALL_LIBEXECDIR=/nix/store/a62mjw98vmi7xznabalpi5yjb43yvchn-snes9x-gtk-1.63/libexec -DCMAKE_INSTALL_LIBDIR=/nix/store/a62mjw98vmi7xznabalpi5yjb43yvchn-snes9x-gtk-1.63/lib -DCMAKE_INSTALL_DOCDIR=/nix/store/a62mjw98vmi7xznabalpi5yjb43yvchn-snes9x-gtk-1.63/share/doc/snes9x-gtk -DCMAKE_INSTALL_INFODIR=/nix/store/a62mjw98vmi7xznabalpi5yjb43yvchn-snes9x-gtk-1.63/share/info -DCMAKE_INSTALL_MANDIR=/nix/store/a62mjw98vmi7xznabalpi5yjb43yvchn-snes9x-gtk-1.63/share/man -DCMAKE_INSTALL_INCLUDEDIR=/nix/store/a62mjw98vmi7xznabalpi5yjb43yvchn-snes9x-gtk-1.63/include -DCMAKE_INSTALL_SBINDIR=/nix/store/a62mjw98vmi7xznabalpi5yjb43yvchn-snes9x-gtk-1.63/sbin -DCMAKE_INSTALL_BINDIR=/nix/store/a62mjw98vmi7xznabalpi5yjb43yvchn-snes9x-gtk-1.63/bin -DCMAKE_INSTALL_NAME_DIR=/nix/store/a62mjw98vmi7xznabalpi5yjb43yvchn-snes9x-gtk-1.63/lib -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_STRIP=/nix/store/dmypp1h4ldn0vfk3fi6yfyf5yxp9yz0k-gcc-wrapper-14.3.0/bin/strip -DCMAKE_RANLIB=/nix/store/dmypp1h4ldn0vfk3fi6yfyf5yxp9yz0k-gcc-wrapper-14.3.0/bin/ranlib -DCMAKE_AR=/nix/store/dmypp1h4ldn0vfk3fi6yfyf5yxp9yz0k-gcc-wrapper-14.3.0/bin/ar -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++ -DCMAKE_INSTALL_PREFIX=/nix/store/a62mjw98vmi7xznabalpi5yjb43yvchn-snes9x-gtk-1.63
-- The C compiler identification is GNU 14.3.0
-- The CXX compiler identification is GNU 14.3.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /nix/store/dmypp1h4ldn0vfk3fi6yfyf5yxp9yz0k-gcc-wrapper-14.3.0/bin/gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /nix/store/dmypp1h4ldn0vfk3fi6yfyf5yxp9yz0k-gcc-wrapper-14.3.0/bin/g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found Gettext: /nix/store/vxhqq9paalnry3qaxlfhiqda9886akd3-gettext-0.25.1/bin/msgmerge (found version "0.25.1")
-- Found PkgConfig: /nix/store/k9s1w95rqfr14lb5xms1lwzrq1yiinzz-pkg-config-wrapper-0.29.2/bin/pkg-config (found version "0.29.2")
-- Checking for module 'sdl2'
--   Found sdl2, version 2.32.56
-- Checking for modules 'gtkmm-3.0;gthread-2.0;libpng'
--   Found gtkmm-3.0, version 3.24.10
--   Found gthread-2.0, version 2.84.4
--   Found libpng, version 1.6.50
Package sysprof-capture-4 was not found in the pkg-config search path.
Perhaps you should add the directory containing `sysprof-capture-4.pc'
to the PKG_CONFIG_PATH environment variable
Package 'sysprof-capture-4', required by 'glib-2.0', not found
Package sysprof-capture-4 was not found in the pkg-config search path.
Perhaps you should add the directory containing `sysprof-capture-4.pc'
to the PKG_CONFIG_PATH environment variable
Package 'sysprof-capture-4', required by 'glib-2.0', not found
Package sysprof-capture-4 was not found in the pkg-config search path.
Perhaps you should add the directory containing `sysprof-capture-4.pc'
to the PKG_CONFIG_PATH environment variable
Package 'sysprof-capture-4', required by 'glib-2.0', not found
Package sysprof-capture-4 was not found in the pkg-config search path.
Perhaps you should add the directory containing `sysprof-capture-4.pc'
to the PKG_CONFIG_PATH environment variable
Package 'sysprof-capture-4', required by 'glib-2.0', not found
-- Checking for module 'xrandr'
--   Found xrandr, version 1.5.4
[33mCMake Warning (dev) at /build/source/external/glslang/CMakeLists.txt:232 (find_package):
  Policy CMP0148 is not set: The FindPythonInterp and FindPythonLibs modules
  are removed.  Run "cmake --help-policy CMP0148" for policy details.  Use
  the cmake_policy command to set the policy and suppress this warning.

Call Stack (most recent call first):
  /build/source/external/glslang/CMakeLists.txt:296 (find_host_package)
This warning is for project developers.  Use -Wno-dev to suppress it.
[0m
-- Found PythonInterp: /nix/store/62fdlzq1x1ak2lsxp4ij7ip5k9nia3hc-python3-3.13.7/bin/python3 (found suitable version "3.13.7", minimum required is "3")
-- spirv-tools not linked - illegal SPIRV may be generated for HLSL
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
[33mCMake Warning (dev) at /build/source/external/glslang/CMakeLists.txt:232 (find_package):
  Policy CMP0148 is not set: The FindPythonInterp and FindPythonLibs modules
  are removed.  Run "cmake --help-policy CMP0148" for policy details.  Use
  the cmake_policy command to set the policy and suppress this warning.

Call Stack (most recent call first):
  /build/source/external/glslang/StandAlone/CMakeLists.txt:34 (find_host_package)
This warning is for project developers.  Use -Wno-dev to suppress it.
[0m
[31mCMake Error at /build/source/external/SPIRV-Cross/CMakeLists.txt:22 (cmake_minimum_required):
  Compatibility with CMake < 3.5 has been removed from CMake.

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.

  Or, add -DCMAKE_POLICY_VERSION_MINIMUM=3.5 to try configuring anyway.

[0m
-- Configuring incomplete, errors occurred!
```

### Additional context

Related to https://github.com/NixOS/nixpkgs/issues/445447 , but not listed in there

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.49-xanmod1, NixOS, 25.11 (Xantusia), 25.11.20251002.7df7ff7`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.31.2`
 - nixpkgs: `/nix/store/9v6qa656sq3xc58vkxslqy646p0ajj61-source`

### Notify maintainers

        @qknight
        @thiagokokada 
        @sugar700

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/sn/snes9x/package.nix
+++ b/pkgs/by-name/sn/snes9x/package.nix
@@ -90,6 +90,11 @@ stdenv.mkDerivation (finalAttrs: {
     ];
 
   postPatch = lib.optionalString withGtk ''
+    # Please remove after snes9x > 1.63.  Fixed by upstream:
+    # https://github.com/snes9xgit/snes9x/commit/a4b4b98fffbde417ad550480021db89f18f11a5d.patch
+    substituteInPlace external/SPIRV-Cross/CMakeLists.txt \
+      --replace-fail 'cmake_minimum_required(VERSION 3.0)' 'cmake_minimum_required(VERSION 3.5)'
+
     substituteInPlace external/glad/src/egl.c \
       --replace-fail libEGL.so.1 "${lib.getLib libGLX}/lib/libEGL.so.1"
     substituteInPlace external/glad/src/glx.c \

